Sources/MockService.swift

```diff
@@ -10,76 +10,166 @@ open class MockService: NSObject {
-              pactVerificationService: PactVerificationService) {
+  ///
+  /// Initializer
+  ///
+  /// - parameter provider: Name of your provider (eg: Calculator API)
+  /// - parameter consumer: Name of your consumer (eg: Calculator.app)
+  /// - parameter pactVerificationService: Your customised `PactVerificationService`
+  ///
+  public init(
+    provider: String,
+    consumer: String,
+    pactVerificationService: PactVerificationService
+    ) {
     self.provider = provider
     self.consumer = consumer
-
     self.pactVerificationService = pactVerificationService
   }
 
+  ///
+  /// Convenience Initializer
+  ///
+  /// - parameter provider: Name of your provider (eg: Calculator API)
+  /// - parameter consumer: Name of your consumer (eg: Calculator.app)
+  ///
+  /// Use this initialiser to use the default PactVerificationService
+  ///
   @objc(initWithProvider: consumer:)
   public convenience init(provider: String, consumer: String) {
     self.init(provider: provider,
               consumer: consumer,
               pactVerificationService: PactVerificationService())
   }
 
+  ///
+  /// Define the providers state
+  ///
+  /// Use this method in the `Arrange` step of your Pact test.
+  ///
+  ///     myMockService.given("a user exists")
+  ///
+  /// - Parameter providerState: A description of providers state
+  /// - Returns: An `Interaction` object
+  ///
   @objc
   open func given(_ providerState: String) -> Interaction {
     let interaction = Interaction().given(providerState)
     interactions.append(interaction)
     return interaction
   }
 
+  ///
+  /// Describe the request your provider will receive
+  ///
+  /// Use this method in the `Arrange` step of your Pact test.
+  ///
+  ///     myMockService.given("a user exists")
+  ///                  .uponReceiving("a request for users")
+  ///
+  /// - Parameter description: Describing the request to the provider
+  /// - Returns: An `Interaction` object
+  ///
   @objc(uponReceiving:)
   open func uponReceiving(_ description: String) -> Interaction {
     let interaction = Interaction().uponReceiving(description)
     interactions.append(interaction)
     return interaction
   }
 
+  ///
+  /// Runs the provided test function with 30 second timeout
+  ///
+  /// Use this method in the `Act` step of your Pact test.
+  /// (eg. Testing your `serviceClientUnderTest!.getUsers(...)` method)
+  ///
+  ///     myMockService!.run { (testComplete) -> Void in
+  ///         serviceClientUnderTest!.getUsers( /* ... */ )
+  ///     }
+  ///
+  /// Make sure you call `testComplete()` after your `Assert` step in your test
+  ///
+  /// - Parameter testFunction: The function making the network request you are testing
+  ///
   @objc(run:)
   open func objcRun(_ testFunction: @escaping (_ testComplete: () -> Void) -> Void) {
     self.run(nil, line: nil, timeout: 30, testFunction: testFunction)
   }
 
+  ///
+  /// Runs the provided test function by specifying timeout in seconds
```

Comment:
This is method is only her for objective-c compatibility so the example should be:

```
[self.mockService run:^(CompleteBlock testComplete) {
     [self. serviceClientUnderTest getUsers]
      testComplete();
} withTimeout: 10];
```